tests/std/tests/VSO_0000000_vector_algorithms/test.cpp

```diff
@@ -776,6 +776,62 @@ void test_swap_ranges(mt19937_64& gen) {
     }
 }
 
+template <class FwdIt, class T>
+FwdIt last_known_good_remove(FwdIt first, FwdIt last, T val) {
+    FwdIt dest = first;
+
+    while (first != last) {
+        if (*first != val) {
+            *dest = *first;
+            ++dest;
+        }
+
+        ++first;
+    }
+
+    return dest;
+}
+
+template <class T>
+void test_case_remove(vector<T>& in_out_expected, vector<T>& in_out_actual, vector<T>& in_out_actual_r, const T val) {
+    auto rem_expected = last_known_good_remove(in_out_expected.begin(), in_out_expected.end(), val);
+    auto rem_actual   = remove(in_out_actual.begin(), in_out_actual.end(), val);
+    assert(equal(in_out_expected.begin(), rem_expected, in_out_actual.begin(), rem_actual));
+
+#if _HAS_CXX20
+    auto rem_actual_r = ranges::remove(in_out_actual_r, val);
+    assert(equal(in_out_expected.begin(), rem_expected, begin(in_out_actual_r), begin(rem_actual_r)));
+#else // ^^^ _HAS_CXX20 / !_HAS_CXX20 vvv
+    (void) in_out_actual_r;
+#endif // ^^^ !_HAS_CXX20 ^^^
+}
+
+template <class T>
+void test_remove(mt19937_64& gen) {
+    using TD = conditional_t<sizeof(T) == 1, int, T>;
+    binomial_distribution<TD> dis(10);
+
+    vector<T> source;
+    vector<T> in_out_expected;
+    vector<T> in_out_actual;
+    vector<T> in_out_actual_r;
+
+    for (auto v : {&source, &in_out_expected, &in_out_actual, &in_out_actual_r}) {

```

Comment:
I know they're pointers but I'd still prefer to see `const auto&` here. Ditto below.